connector_ecommerce/components/sale_order_onchange.py

```diff
@@ -93,8 +95,21 @@ def play(self, order, order_lines):
             for idx, command_line in enumerate(line_list):
                 # line_list format:[(0, 0, {...}), (0, 0, {...})]
                 if command_line[0] in (0, 1):  # create or update values
+                    # we work on a temporary record
+                    order_model = self.env["sale.order"]
+
+                    all_values = {
+                        k: v for k, v in order.items() if k in order_model._fields
+                    }
+                    for field in order_model._fields:
+                        if field not in all_values:
+                            all_values[field] = False
+                    new_record = order_model.new(all_values)
                     # keeps command number and ID (or 0)
                     old_line_data = command_line[2]
+                    # passing order_id also so order_id.company_id based onchange will
+                    # also work
+                    old_line_data["order_id"] = new_record
```

Comment:
Shouldn't this be like this?
```suggestion
                    old_line_data["order_id"] = new_record.id
```